backend/homework/1/app/src/main/java/org/JFPart1/Main.java

```diff
@@ -0,0 +1,119 @@
+            System.out.println("1. Add new Student to repository");
+            System.out.println("2. Update a Student information");
+            System.out.println("3. Retrieve any Student information");
+            System.out.println("4. Retrieve all Student information");
+            System.out.println("5. Close program");
+
+            int option = checkNumber(sc.nextLine());
+            Student student;
+            boolean isSuccessful = false;
+
+            switch(option){
+                case 1:
+                    student = inputStudent();
+                    isSuccessful = studentRepository.addStudent(student);
+                    if(isSuccessful)loggingClass.addedSuccessfully(student);
+                    else loggingClass.repeatedStudent(student);
+                    break;
+
+                case 2:
+                    student = inputStudent();
+                    isSuccessful = studentRepository.updateStudent(student);
+                    if (isSuccessful)loggingClass.updatedStudent(student);
+                    else loggingClass.studentNotFound(student.getId());
+                    break;
+
+                case 3:
+                    System.out.println("\n\nEnter how do you wanna retrieve student,");
+                    System.out.println("1 for using Student id");
+                    System.out.println("2 for using Student name");
+                    int input = checkNumber(sc.nextLine());
+                    if(input == 1){
+                        System.out.print("Enter id: ");
+                        int id = checkNumber(sc.nextLine());
+                        Student s = studentRepository.getStudent(id);
+                        if(s == null)loggingClass.studentNotFound(id);
+                        else {
+                            System.out.println("Id: " + s.getId() + ", Name: " + s.getName() + ", Age: " + s.getAge() + ", Grade: " + s.getGrade());
+                        }
+                    }
+                    else {
+                        System.out.print("Enter name: ");
+                        String name = sc.nextLine();
+                        List<Student> studentList = studentRepository.getStudent(name);
+                        if(studentList == null)loggingClass.studentNotFound(name);
+                        else{
+                            studentList.forEach(s -> System.out.println("Id: " + s.getId() + ", Name: " + s.getName() + ", Age: " + s.getAge() + ", Grade: " + s.getGrade()));
+                        }
+                    }
+                    break;
+
+                case 4:
+                    System.out.println("This is the list of all students,\n");
+                    studentRepository.getAllStudents().forEach(s -> {
+                        System.out.println("Id: " + s.getId() + ", Name: " + s.getName() + ", Age: " + s.getAge() + ", Grade: " + s.getGrade());
+                    });
+                    System.out.println("\n\n");
```

Comment:
Again logic is too tighly coupled here. Move it out in a function try to make it more readable. Add comments and try to make each function implement a single logic.